src/schema_compile_node.c

```diff
@@ -1185,35 +1185,35 @@ lys_compile_type_enums(struct lysc_ctx *ctx, struct lysp_type_enum *enums_p, LY_
 
         if (basetype == LY_TYPE_ENUM) {
             if (enums_p[u].flags & LYS_SET_VALUE) {
+                /* value assigned by model */
                 cur_val = (int32_t)enums_p[u].value;
-                if (!u || (cur_val >= value)) {
-                    value = cur_val + 1;
-                }
-                /* check collision with other values */
-                LY_ARRAY_FOR(*enums, v) {
-                    if (cur_val == (*enums)[v].value) {
-                        LOGVAL(ctx->ctx, LY_VLOG_STR, ctx->path, LYVE_SYNTAX_YANG,
-                                "Invalid enumeration - value %d collide in items \"%s\" and \"%s\".",
-                                cur_val, enums_p[u].name, (*enums)[v].name);
-                        return LY_EVALID;
-                    }
-                }
             } else if (base_enums) {
                 /* inherit the assigned value */
                 cur_val = base_enums[match].value;
-                if (!u || (cur_val >= value)) {
-                    value = cur_val + 1;
-                }
             } else {
                 /* assign value automatically */
-                if (u && (value == INT32_MIN)) {
-                    /* counter overflow */
+                if (u == 0) {
+                    cur_val = 0;
+                }
```

Comment:
This block seems redundant to me but some refactorization of the `if` block is needed so that the `0` is kept (or do not initialize `cur_val`).